### DISCUSSION TITLE
Pattern matching on source

### DISCUSSION THREAD

**Discussion by jinzhongjia** [hrsh7th/nvim-cmp#1421] [Q&A] [answered]
Now, I'm write source for Codegeex( a ai complete tool )

The default completion display rule is prefix match?

The result given by codegeex may not match the prefix, how can I make it always appear in the completion list?

**Comment by hrsh7th** [comment, marked as answer]
You can use `filterText` property that can be specifeid the CompletionItem type.

You should learn [LSP](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#completionItem) specification.

**Reply by hrsh7th** [reply]
The nvim-cmp's default matching strategy is not the prefix match.
It's fuzzy match by default.

**Reply by jinzhongjia** [reply]
thx
